allennlp/semparse/worlds/atis_world.py

```diff
@@ -31,74 +59,182 @@ def __init__(self,
                  utterances: List[str],
                  tokenizer: Tokenizer = None,
                  database_directory: str = None) -> None:
+
+        self.all_tables = ALL_TABLES
+        self.tables_with_strings = TABLES_WITH_STRINGS
         if database_directory:
-            self.sql_table_context = SqlTableContext(ALL_TABLES,
-                                                     TABLES_WITH_STRINGS,
-                                                     database_directory)
+            self.database_directory = database_directory
+            self.connection = sqlite3.connect(self.database_directory)
+            self.cursor = self.connection.cursor()
+
+        self.sql_table_context = SqlTableContext(ALL_TABLES,
+                                                 TABLES_WITH_STRINGS,
+                                                 database_directory,
+                                                 utterances=utterances) if database_directory else None
+
+        self.grammar_dictionary = deepcopy(self.sql_table_context.grammar_dictionary)
```

Comment:
I don't think you need this `deepcopy`, because you're instantiating one `SqlTableContext` per `AtisWorld`, and the `SqlTableContext` already did a `deepcopy`.